### PR TITLE
9904 Fix double quoting default values in Kotlin Spring

### PR DESCRIPTION
**Pull request by bamapookie**
Fix for #9904.

Kotlin files with default values that are not enums were getting double quoted (`defaultValue = ""default""`) in API file generation.  This fixes the issue for query parameters.  

The first commit adds a method to the petstore sample that uses the code path demonstrating this bug.  See the second commit for the invalid source generated.  (kotlin-springboot)

CC: Kotlin committee members @jimschubert, @dr4ke616, @karismann, @Zomzog, @andrewemery, @4brunu, and @yutaka0m

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Fix should go into 5.4.0 branch as well.

### CODE REVIEW

**Comment by bamapookie** [comment]
I'm unsure what to do here.  There are tests failing for go and jaxrs, but I don't know why.  I added a new endpoint to [openapi-generator/src/test/resources/2_0/petstore.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/2_0/petstore.yaml), because that is the file that generated the samples, but the documentation in [CONTRIBUTING.md](https://github.com/OpenAPITools/openapi-generator/blob/master/CONTRIBUTING.md) says to update [openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml).  Which one should I update?

File: modules/openapi-generator/src/test/resources/2_0/petstore.yaml [hunk 1]
```diff
@@ -112,6 +112,37 @@ paths:
...
+            type: array
+            items:
+              $ref: '#/definitions/Pet'
+        '400':
+          description: Invalid name value
+      security:
+        - petstore_auth:
+            - 'write:pets'
+            - 'read:pets'
   /pet/findByTags:
```
Comment (wing328) [inline modules/openapi-generator/src/test/resources/2_0/petstore.yaml:146]: Please undo changes to petstore.yaml. We'll create a separate test spec for Kotlin Spring generator and will include this particular test.

**Comment by wing328** [comment]
For the Go related error in CircleCI node 2, please ignore that for the time being.

File: modules/openapi-generator/src/test/resources/2_0/petstore.yaml (hunk 1 above)
Comment (bamapookie) [reply modules/openapi-generator/src/test/resources/2_0/petstore.yaml]: Removed the petstore.yaml change and its related sample changes.

**Comment by bamapookie** [comment]
Rebased on current upstream master.

**Comment by wing328** [comment]
cc @jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

**Comment by wing328** [comment]
Tested locally to confirm the fix. Thanks for the PR.

I'll try to add some tests later to cover this.

**Pull request merged**